libraries/Microsoft.Bot.Builder.Azure/CosmosDbPartitionedStorage.cs

```diff
@@ -381,41 +381,63 @@ private async Task CreateContainerIfNotExistsAsync()
         /// <summary>
         /// Internal data structure for storing items in a CosmosDB Collection.
         /// </summary>
-        private class DocumentStoreItem : IStoreItem
```

Comment:
I'm concerned about moving this from it's currently nested position within `CosmosDbPartitionedStorage` due to possible typing serialization issues.